drake/systems/plants/inverseDynamicsmex.cpp

```diff
@@ -24,21 +24,24 @@ void mexFunction(int nlhs, mxArray *plhs[],int nrhs, const mxArray *prhs[]) {
   int gradient_order = nlhs - 1;
 
 
-  RigidBodyManipulator *model = (RigidBodyManipulator*) getDrakeMexPointer(prhs[0]);
+  int arg_num = 0;
+  RigidBodyManipulator *model = static_cast<RigidBodyManipulator*>(getDrakeMexPointer(prhs[arg_num++]));
+  KinematicsCache<double>* cache = static_cast<KinematicsCache<double>*>(getDrakeMexPointer(prhs[arg_num++]));
+
   int nq = model->num_positions;
   int nv = model->num_velocities;
   const mxArray* f_ext_matlab = nullptr;
   const mxArray* vd_matlab = nullptr;
   const mxArray* df_ext_matlab = nullptr;
   const mxArray* dvd_matlab = nullptr;
-  if (nrhs > 1)
-    f_ext_matlab = prhs[1];
   if (nrhs > 2)
-    vd_matlab = prhs[2];
+    f_ext_matlab = prhs[arg_num++];
   if (nrhs > 3)
```

Comment:
I bet apple wishes they hadn't allowed the no-brace style :wink: 

https://www.imperialviolet.org/2014/02/22/applebug.html